app/services/flow/flow_state_machine.rb

```diff
@@ -137,6 +142,10 @@ def analytics_visited
       @analytics_id + ' visited'
     end
 
+    def analytics_optional_step
+      @analytics_id + 'optional submitted'
```

Comment:
needs a space here in the beginning of `optional submitted` like the one above

Maybe less error-prone as:

```suggestion
      [@analytics_id, 'optional submitted'].join(' ')
```